app/plugin_key_value_store.go

```diff
@@ -68,6 +68,28 @@ func (a *App) CompareAndSetPluginKey(pluginId string, key string, oldValue, newV
 	return updated, nil
 }
 
+func (a *App) CompareAndSetPluginKeyWithExpiry(pluginId string, key string, oldValue, newValue []byte, expireInSeconds int64) (bool, *model.AppError) {
+	kv := &model.PluginKeyValue{
+		PluginId: pluginId,
+		Key:      key,
+		Value:    newValue,
+		ExpireAt: expireInSeconds,
```

Comment:
Given that this `CompareAndSetPluginKeyWithExpiry` has the exact same code as `CompareAndSetPluginKey` with the only difference of this line where the `ExpireAt` is set, I think it's best to have `CompareAndSetPluginKey` call this method `CompareAndSetPluginKeyWithExpiry` passing 0 for `expireInSeconds`.  This way we avoid code duplication and it's also consistent with the implementation for [SetPluginKey](https://github.com/mattermost/mattermost-server/blob/fe8a0f64856ff93e0a8125e6fd134d682453321e/app/plugin_key_value_store.go#L21)

Reply:
Done.